provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -471,12 +461,12 @@ static int process_riot_key_info(HSM_CLIENT_X509_INFO* riot_info)
                 LogError("Failure: producing alias cert.");
                 result = MU_FAILURE;
             }
-            else if (generate_root_ca_info(riot_info, &tbs_sig) != 0)
+            else if (generate_root_ca_info(riot_info) != 0)
```

Comment:
I see that the root/device functions calculate the to-be-signed data internally. 
Why do we have this difference (and extra computing)?

Reply:
This is where the memory leak was occurring, before we would pass the TBS pointer through, however since the new crypto lib has an mpi_grow function that allocates we were losing a corresponding free, because Riot_crypt_sign takes the TBS and re-inits its members before we can free the previous pointers, I tried a lot of different approaches to mitigate this and this was the one that worked, if we don't like this approach I could go back and try the pointer passing again

Reply:
When this was explained to me it was in the context of mpi_grow.  There is no issue with mpi_grow.  Now that I read this comment it is clear that yes, we do drop two MPIs, s and r.  This is now fixed.

Reply:
Last update to RIoT was 14 days ago, I'm holding on this until I see a change

Reply:
@DMattoon is this change reflected in RIoT master? or are we still required to handle this in our code?